mitodl/mitxonline
# Support enrolling learner as verified on payment

#### Pre-Flight checklist

  - [?] Code is tested

#### What are the relevant tickets?
Fixes #439
Fixes #434

#### What's this PR do?
Broadly, this PR enrolls the learner with verified mode and redirects them to the dashboard with an alert message. In the case of failures, they're redirected back to the cart with an appropriate error message. 

To support all of this I made the following changes:

- Consolidated the callback code into a single spot.
- Refactored a few things out of the views file into `api.py`
- Moved the clearing of the cart under after the order is complete to support failure modes better.
- Moved some of the logic that was in views into the data models.
- Added recording of the cybersource transaction.
- Fixed up some rough edges with order data models
- Added additional states to the order model to bring it to parity with the states from CyberSource
- Removed test views (cleanup)

#### How should this be manually tested?

- Add a course  to your cart
- Go to check out
- Click "cancel order" at the bottom 
- You should get directed back to the cart and your product is still present
- Go to checkout again
- Complete the payment
- Verify that you end up back at the dashboard with a confirmation message below and your cart is now cleared.

#### Screenshots (if appropriate)
![Screenshot 2022-03-02 at 17-54-43 mitX Online Dashboard](https://user-images.githubusercontent.com/28598/156464569-68080850-4ada-405a-a46a-5d71e515b145.png)
